### DISCUSSION TITLE
Need a little help designing my configuration

### DISCUSSION THREAD

**Discussion by Woefdram** [rspamd/rspamd#4669] [General]
After having installed rspamd and seeing it filter my incoming mail like it should (mostly), I have to also configure outgoing mail, which should be treated quite differently. But I don't really see how I can do this.

I'm looking for some general hints, experience from people who've already done something like this.

### Background

My production setup consists of 3 types of servers: SMTP; Scanner; IMAP/Submission. The scanner accepts mail from both sides, scans it and hands it over to the next hop. Because rspamd has to be a drop-in replacement, I need it to behave (mostly) like Amavis does now.

My current scanner, Amavis, knows 2 policies, "incoming" and "outgoing", both defining exactly what should happen with a message. I think I want something like that in rspamd too, but this is where I'd like some suggestions/hints from people who have more experience with rspamd than I.

The architecture dictates that every mail coming in from the SMTP side, must always be accepted, scanned (post-queue, ideally), adorned with a few headers and then delivered to IMAP over LMTP. Unless it contains a virus, in which case it must be quarantined and the receiver(s) notified. An incoming message can never ever be refused, because that would mean the SMTP-server has to send a bounce-message for an already accepted message.

Outgoing mail however should be scanned before-queue, and immediately rejected if it smells bad. The user submitting the message gets an immediate notification that his messages was not accepted.

### Where to begin?

How can I let rspamd discern between incoming and outgoing mail? Or maybe better: how do I let rspamd recognise authenticated users without anybody ever logging in on the scanner?

An alternative to configuring rspamd bidirectionally, could be to make 2 installations: one for incoming and one for outgoing mail. But that doesn't feel right, to be honest.

I'm sure I'm not the first one who needs a kind of standalone scanner, integrated in an existing mail environment. How have you people tackled this?

**Comment by fatalbanana** [comment]
Use [settings](https://rspamd.com/doc/configuration/settings.html) (to conditionally disable `reject` etc); refer to hints about [scanning outbound](https://rspamd.com/doc/tutorials/scanning_outbound.html).

Rspamd knows usernames (and so that someone authenticated) via milter / other suitable integration.

Rspamd doesn't offer to deal with deliveries, they need to be handled in MTA; integration is generally limited to accepting/rejecting a message & adding/removing headers.
